packages/core/src/hooks/estimateGasLimit.test.ts

```diff
@@ -0,0 +1,36 @@
+import { expect } from 'chai'
+import { MockProvider } from 'ethereum-waffle'
+import { estimateGasLimit } from './usePromiseTransaction'
+
+const BASE_TX_COST = 21000
+
+describe('estimateGasLimit', () => {
```

Comment:
Not exhaustive set of tests, missing:

- user specifying custom gas limit to the transaction, with buffer
- user specifying custom gas limit to the transaction, without buffer